esmvalcore/_recipe/check.py

```diff
@@ -391,20 +400,25 @@ def _check_format_years(date):
 
 
 def _check_timerange_values(date, timerange):
+    # P must always be in a duration string
+    # if T in date, that is a datetime; otherwise it's date
     try:
-        isodate.parse_date(date)
-    except ValueError:
-        try:
+        if "P" in date:
             isodate.parse_duration(date)
-        except ValueError as exc:
-            if date != "*":
-                raise RecipeError(
-                    "Invalid value encountered for `timerange`. "
-                    "Valid value must follow ISO 8601 standard "
-                    "for dates and duration periods, or be "
-                    "set to '*' to load available years. "
-                    f"Got {timerange} instead."
-                ) from exc
+        else:
+            if "T" in date:
+                isodate.parse_datetime(date)
```

Comment:
```py
if "P" in data:
    isodate.parse_duration(date)
elif "T" in date:
    isodate.parse_datetime(date)
else:
    isodate.parse_date(date)
```

should also work

Reply:
ace, Manu! Plugged in in 7f631a14350cd94004cbf91d6c989f1bad09ff17